packages/ra-core/src/dataProvider/fetch.ts

```diff
@@ -14,9 +14,12 @@ export const createHeadersFromOptions = (options: Options): Headers => {
             Accept: 'application/json',
         })) as Headers;
     if (
+        // An application/json Content-Type header in requests without a body breaks some parsers, like fastify
+        options &&
+        options.body &&
         !requestHeaders.has('Content-Type') &&
-        !(options && (!options.method || options.method === 'GET')) &&
-        !(options && options.body && options.body instanceof FormData)
+        !(!options.method || options.method === 'GET') &&
+        !(options.body instanceof FormData)
```

Comment:
The accumulation of conditions and the double negations make it hard to read. Could you please make the code clearer? Something like:

```jsx
const shouldSetContentType = () => {
        const hasBody = options && options.body;
        const isContentTypeSet = requestHeaders.has('Content-Type');
        const isGetMethod = !options?.method || options?.method === 'GET';
        const isFormData = options?.body instanceof FormData;

        return hasBody && !isContentTypeSet && !isGetMethod && !isFormData;
    };
```

Reply:
right! I did just that